ee/tpmrunner/tpmrunner.go

```diff
@@ -67,7 +69,16 @@ func New(ctx context.Context, slogger *slog.Logger, store types.GetterSetterDele
 	}
 
 	// assume we have a tpm until we know otherwise
-	tpmRunner.machineHasTpm.Store(true)
+	hasTPM := true
+
+	// on linux the TPM is at /dev/tpm0
+	// if it doesn't exist, we don't have a TPM
+	if runtime.GOOS == "linux" {
+		_, err := os.Stat("/dev/tpm0")
+		hasTPM = err == nil
```

Comment:
since this is a one time check maybe worth logging the hasTPM verdict + err here?

Reply:
good call! done